frontend/src/metabase/visualizations/visualizations/WaterfallChart.jsx

```diff
@@ -53,6 +55,21 @@ export default class WaterfallChart extends LineAreaBarChart {
 
   static renderer = waterfallRenderer;
 
+  static checkRenderable = (series, settings) => {
+    super.checkRenderable(series, settings);
+
+    const [{ card }] = series;
+    const question = new Question(card);
+    // query().breakouts() does not exist in native queries
+    const queryBreakouts = question.query().breakouts?.() || [];
```

Comment:
99% chance this is incorrect way to implement this check and it won't work with any real use case (e.g. multi-stage queries etc.). Should you check for `settings["graph.dimensions"]` instead?

Reply:
Also as mentioned this check won't work for native queries

Reply:
Checking `settings["graph.dimensions"]` would work for all cases

Reply:
The problem is that checking for graph.dimensions will not be sufficent, because the series has already been affected. In this screenshot, you can see that there are duplicates sources in the series data, even though graph.dimensions is only `'SOURCE`".

Maybe the better thing to do would just be to fix the graph

**
![image](https://user-images.githubusercontent.com/1328979/193827581-2d840aa8-89db-4ef4-a879-26831ba97f84.png)
**